mara_db/format.py

```diff
@@ -0,0 +1,53 @@
+"""Different formats for piping"""
+
+
+class Format:
+    """Base format definition"""
+
+    def __repr__(self) -> str:
+        return (f'<{self.__class__.__name__}: '
+                + ', '.join([f'{var}={getattr(self, var)}'
+                             for var in vars(self) if getattr(self, var)])
+                + '>')
+
+
+class CsvFormat(Format):
+    """
+    CSV file format. See https://tools.ietf.org/html/rfc4180
+    """
+    def __init__(self, delimiter_char: str = None, quote_char: str = None, header: bool = None):
+        """
+        CSV file format. See https://tools.ietf.org/html/rfc4180
+
+        Args:
+            delimiter_char: The character that separates columns
+            quote_char: The character for quoting strings
+            header: Whether a csv header with the column name(s) is part of the CSV file.
+        """
+        self.delimiter_char = delimiter_char
+        self.quote_char = quote_char
+        self.header = header
+
```

Comment:
Do you see a chance to make the validation of the parameter part of this? E.g. we have "setting a default" and "prevent bad settings", so maybe something like 

```
def set_default(*, delimiter_char: Optional[bool] = None,...)
    if delimiter_char is not None and self.delimiter_char is None:
        self.delimiter_char = delimiter_char
    ...
def validate(*, *, allowed_delimiter_char_values: Optional[Set[bool|None]] = None,..):
    if allowed_delimiter_char_values is not None:
        if self.delimiter_char not in allowed_delimiter_char_values:
            raise RuntimeError("....)
    ...
```

Reply:
Done